### PR TITLE
Shell creates a dynamic window with a webview2 control

### PR DESCRIPTION
**Pull request by kruplm**

### CODE REVIEW

File: Tryouts/Prototypes/Shell/WebContent.xaml.cs
Comment (ztanczos) [inline Tryouts/Prototypes/Shell/WebContent.xaml.cs:25]: WebContent is also a Window, you don't need to create another one.

File: Tryouts/Prototypes/Shell/MainWindow.xaml.cs
Comment (ztanczos) [inline Tryouts/Prototypes/Shell/MainWindow.xaml.cs:25]: Same comment here

File: Tryouts/Prototypes/Shell/MainWindow.xaml.cs
Comment (ztanczos) [inline Tryouts/Prototypes/Shell/MainWindow.xaml.cs:34]: Creating and showing a window are two separate actions. I wouldn't show a window from the ctor, instead the caller should decide when/how to show it.

File: Tryouts/Prototypes/Shell/WebContent.xaml.cs
Comment (ztanczos) [inline Tryouts/Prototypes/Shell/WebContent.xaml.cs:61]: _webView2 is guaranteed to be non-null

**Review by BalassaMarton — APPROVED** [review]

**Comment by ZKRobi** [comment]
Just a reminder, please squash commits before merging the PR :)

File: Tryouts/Prototypes/Shell/Manifest/ManifestParser.cs
Comment (ztanczos) [inline Tryouts/Prototypes/Shell/Manifest/ManifestParser.cs:28]: Properties should be PascalCased

File: Tryouts/Prototypes/Shell/Manifest/ManifestParser.cs
Comment (ztanczos) [inline Tryouts/Prototypes/Shell/Manifest/ManifestParser.cs:44]: An empty catch block doesn't add any value - you can remove the try/catch from here

File: Tryouts/Prototypes/Shell/Manifest/ManifestParser.cs
Comment (ztanczos) [inline Tryouts/Prototypes/Shell/Manifest/ManifestParser.cs:42]: There's also an overload which takes a Stream instead of a string. You might want to consider using that instead.

File: Tryouts/Prototypes/Shell/MainWindow.xaml.cs
Comment (ztanczos) [inline Tryouts/Prototypes/Shell/MainWindow.xaml.cs:73]: This window should not assume how many modules were parsed from the manifest.

File: Tryouts/Prototypes/Shell/Manifest/ManifestParser.cs
Comment (BalassaMarton) [inline Tryouts/Prototypes/Shell/Manifest/ManifestParser.cs:37]:
```cs
using var stream = File.OpenRead(manifestFile);
manifest = JsonSerializer.Deserialize<ManifestModel>(stream, ManifestModel.JsonSerializerOptions);
```

File: Tryouts/Prototypes/Shell/Manifest/ModuleModel.cs
Comment (BalassaMarton) [inline Tryouts/Prototypes/Shell/Manifest/ModuleModel.cs:21]: No need for this attribute, it's only for legacy remoting (SOAP mostly).

**Pull request merged**
